### DISCUSSION TITLE
Can finn accelerate 4w4a or 8w8a models ?

### DISCUSSION THREAD

**Discussion by YangNuoCheng** [Xilinx/finn#578] [Q&A]
## Can finn accelerate 4w4a or 8w8a models ?
I noticed there no `CNV_4w4a`  and ` CNV_8w8a` data in [TABLE](https://docs.google.com/spreadsheets/d/14W4PcQonAuFLl5Vv7xxyUIIhFM6IPIiTuSfNULvEl4c/edit#gid=441390913)
is the FINN didn't support 4bit accelerate?I got the accelerate idea from [FINN-R](https://dl.acm.org/doi/10.1145/3242897).

I modified ` BNN_PYNQ`  in ` brevitas_examples`  to get `CNV_2w2a` 、`CNV_4w4a`  and ` CNV_8w8a`  modules and use following code for export.
````
from brevitas.export import FINNManager
FINNManager.export(self.model, input_shape=(1, 3, 32, 32), export_path=self.args.network+".onnx")
````
I'm tring to accelerate the quantized model on PYNQ-Z2 board.
When I using `cnv=get_test_model_trained("CNV",2,2))`  or my export CNV_2w2a.onnx there have same bugs as 'https://github.com/Xilinx/finn/discussions/437' (InferQuantizedStreamingFCLayer: Signed output requres actval < 0) in `Lowering and Streamlining` part.
But I didn't modified the construction of CNV.
How can I get the same CNV_2w2a result from  [TABLE](https://docs.google.com/spreadsheets/d/14W4PcQonAuFLl5Vv7xxyUIIhFM6IPIiTuSfNULvEl4c/edit#gid=441390913)？

**Comment by YangNuoCheng** [comment]
here are my exported onnx and notebook based on BNN-PYNQ
[CNV.zip](https://github.com/Xilinx/finn/files/8341093/CNV.zip)

**Reply by Ba1tu3han** [reply]
Hello @YangNuoCheng ,

Have you solved it? Could you explain it please?

Bests,
